src/hdmf/spec/namespace.py

```diff
@@ -83,13 +85,16 @@ def contact(self):
 
     @property
     def author(self):
-        """String or list of strings with the authors or  None"""
+        """String or list of strings with the authors or None"""
         return self.get('author', None)
 
     @property
     def version(self):
-        """String, list, or tuple with the version or None """
-        return self.get('version', None)
+        """
+        String, list, or tuple with the version or SpecNamespace.UNVERSIONED
+        if the version is missing or empty
+        """
+        return self.get('version', None) or SpecNamespace.UNVERSIONED
```

Comment:
This would return a bool. Did you mean this?

`return self.get('version', SpecNamespace.UNVERSIONED)`

Reply:
Interestingly, this does not return a bool. 
```python                            
>>> 'a' or 'b'
'a'
>>> False or 'b'
'b'
>>> None or 'b'
'b'
>>> None or None
>>>                      
```
As written, if `version` is somehow '', [], or (), then `SpecNamespace.UNVERSIONED` will be returned. 

If `self.get('version', SpecNamespace.UNVERSIONED)` is used, then if `version` is somehow '', [], or (), then those would get returned.